jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -529,7 +547,22 @@ public Builder withBaggageRestrictionManager(BaggageRestrictionManager baggageRe
     }
 
     public Tracer build() {
-      return new Tracer(this.serviceName, reporter, sampler, registry, clock, metrics, tags,
+      if (serviceName == null) {
+        serviceName = "unknown";
+        log.warn("Service name hasn't been configured, using " + serviceName + " instead");
```

Comment:
What is the argument for allowing "unknown"? Service name is a fundamental concept in Jaeger, I would rather require it in the `builder()` func.

Reply:
+1 there si a comment in a PR description about this